File: Beakjoon/src/Silver/Lv4/boj_1931.java
package Silver.Lv4;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.Arrays;
import java.util.Comparator;

public class boj_1931 { // 회의실 배정
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int N = Integer.parseInt(br.readLine());
        int[][] arr = new int[N][2];

        for (int i = 0; i < N; i++) {
            String[] inputs = br.readLine().split(" ");

            arr[i][0] = Integer.parseInt(inputs[0]);
            arr[i][1] = Integer.parseInt(inputs[1]);
        }

        Arrays.sort(arr, new Comparator<int[]>() {
            public int compare(int[] o1, int[] o2) {
                if (o1[1] == o2[1]) return o1[0] - o2[0];
                else return o1[1] - o2[1];
            }
        });

        int end = 0;
        int count = 0;

        for (int i = 0; i < N; i++) {
            if (end <= arr[i][0]) {
                end = arr[i][1];
                count++;
            }
        }

        bw.write(count + "\n");
        bw.flush();
        bw.close();
    }
}

File: Beakjoon/src/Silver/Lv4/boj_1931.kt
package Silver.Lv4

data class meeting(var start: Int, var end: Int)

fun main(args: Array<String>) {  // 회의실 배정
    val N = readln().toInt()
    var arr = Array<meeting?>(N) { null }

    for (i in 0 until N) {
        val input = readln().split(" ").map { it.toInt() }
        val people = meeting(input[0], input[1])

        arr[i] = people
    }

    arr.sortWith(compareBy({ it!!.end }, { it!!.start }))
    var end = 0
    var count = 0


    arr.forEach {
        if (end <= it!!.start) {
            end = it.end
            count++
        }
    }

    println(count)
}